docs/quadratic_programming.rst

```diff
@@ -0,0 +1,316 @@
+
+CvxpyQP
+~~~~~~~
+
+The wrapper over Cvxpy is a high precision solver that runs in ``float64``.
+However it is not jittable, and does not support matvec and pytrees.
+
+.. autosummary::
+  :toctree: _autosummary
+
+    jaxopt.CvxpyQP
+
+Example::
+
+  from jaxopt import CvxpyQP
+
+  Q = 2 * jnp.array([[2.0, 0.5], [0.5, 1]])
+  c = jnp.array([1.0, 1.0])
+  A = jnp.array([[1.0, 1.0]])
+  b = jnp.array([1.0])
+  G = jnp.array([[-1.0, 0.0], [0.0, -1.0]])
+  h = jnp.array([0.0, 0.0])
+
+  qp = CvxpyWrapper()
+  sol, _ = qp.run(params_obj=(Q, c), params_eq=(A, b), params_ineq=(G, h)).params
+
+  print(sol.primal)
+  print(sol.dual_eq)
+  print(sol.dual_ineq)
+
+Only one of ``params_eq`` and ``params_ineq`` are is mandatory.
+
+OSQP
+~~~~
+
+This solver is a pure Jax re-implementation of OSQP algorithm.
+It is jittable, supports pytrees and matvecs, but the precision is usually lower than :class:`CvxpyQP`.
+
+.. autosummary::
+  :toctree: _autosummary
+
+    jaxopt.OSQP
+
+Example::
+
+  from jaxopt import OSQP
+
+  Q = 2 * jnp.array([[2.0, 0.5], [0.5, 1]])
+  c = jnp.array([1.0, 1.0])
+  A = jnp.array([[1.0, 1.0]])
+  b = jnp.array([1.0])
+  G = jnp.array([[-1.0, 0.0], [0.0, -1.0]])
+  h = jnp.array([0.0, 0.0])
+
+  qp = OSQP()
+  sol, state = qp.run(params_obj=(Q, c), params_eq=(A, b), params_ineq=(G, h)).params
+
+  print(sol.primal)
+  print(sol.dual_eq)
+  print(sol.dual_ineq)
+
+See :class:`BoxOSQP` for a full description of parameters.
+
+Box constraints QPs
+-------------------
+
+The problem takes the form:
+
+.. math::
+
+    \min_{x,z} \frac{1}{2} x^\top Q x + c^\top x \textrm{ subject to } A x = z, l \le z \le u
+
+.. autosummary::
+  :toctree: _autosummary
+
+    jaxopt.BoxOSQP
+
+:class:`BoxOSQP` is similar to :class:`OSQP` but accepts this input format instead.
+The bounds ``u`` (resp. ``l``) can be set to ``inf`` (resp. ``-inf``) if required.
+
+
+Example::
+
+  from jaxopt import BoxOSQP
+
+  Q = 2 * jnp.array([[2.0, 0.5], [0.5, 1]])
+  c = jnp.array([1.0, 1.0])
+  A = jnp.array([[1.0, 1.0], [-1.0, 0.0], [0.0, -1.0]])
+  l = jnp.array([1.0, -jnp.inf, -jnp.inf])
+  u = jnp.array([1.0, 0.0, 0.0])
+
+  qp = BoxOSQP()
+  sol, state = qp.run(params_obj=(Q, c), params_eq=A, params_ineq=(l, u)).params
+
+  print(sol.primal)
+  print(sol.dual_eq)
+  print(sol.dual_ineq)
+
+If required the algorithm can be speed up by setting ``check_primal_dual_infeasability`` to ``False``,
+by setting ``eq_qp_preconditioner`` to ``"jacobi"`` (when possible).  
+  
+.. note::
+
+  The ``tol`` parameter is used as stopping criterion when compared to primal and dual residuals.
+  For over constrained problems, or badly scaled matrices, the residuals can be high, and the solver might
+  never converge up to the desired accuracy.
+  In this case it is better to increases ``tol`` or to reduce ``maxiter``.  
+
+Unconstrained QPs
+-----------------
+
+.. autosummary::
+  :toctree: _autosummary
+
+    jaxopt.linear_solve.solve_cg
```

Comment:
I think it's better to not list it, as it's a tool to solve an unconstrained QP but not a QP solver per se.